src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/Extensions/CastExpressionSyntaxExtensions.vb

```diff
@@ -10,21 +10,112 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
+                Dim foundEOL As Boolean = False
+                For i As Integer = 0 To trailingTrivia.Count - 1
+                    Dim trivia As SyntaxTrivia = trailingTrivia(i)
+                    Dim nextTrivia As SyntaxTrivia = GetForwardTriviaOrDefault(trailingTrivia, i)
+                    Select Case trivia.Kind
+                        Case SyntaxKind.WhitespaceTrivia
+                            Select Case nextTrivia.Kind
+                                Case SyntaxKind.WhitespaceTrivia
+                                    trailingTrivia(i + 1) = AdjustWhitespace(trivia, nextTrivia, GetForwardTriviaOrDefault(trailingTrivia, i, lookaheadCount:=2).IsKind(SyntaxKind.LineContinuationTrivia))
+                                Case SyntaxKind.EndOfLineTrivia
+                                    ' skip Whitespace before EOL
+                                Case SyntaxKind.LineContinuationTrivia
+                                    If GetForwardTriviaOrDefault(trailingTrivia, i, lookaheadCount:=2).IsKind(SyntaxKind.WhitespaceTrivia) Then
+                                        trailingTrivia(i + 2) = trivia
+                                        trivia = SyntaxFactory.Space
+                                    End If
+                                    newTrailingTrivia.Add(trivia)
+                                Case SyntaxKind.CommentTrivia
+                                    newTrailingTrivia.Add(trivia)
+                                Case Else
+                                    newTrailingTrivia.Add(trivia)
+                            End Select
+                        Case SyntaxKind.EndOfLineTrivia
+                            If Not foundEOL Then
+                                newTrailingTrivia.Add(trivia)
+                                foundEOL = True
+                            End If
+                        Case SyntaxKind.LineContinuationTrivia
+                            newTrailingTrivia.Add(trivia)
+                        Case SyntaxKind.CommentTrivia
+                            newTrailingTrivia.Add(trivia)
+                            foundEOL = False
+                        Case Else
+                            foundEOL = False
+                            newTrailingTrivia.Add(trivia)
+                    End Select
+                Next
+                trailingTrivia = newTrailingTrivia
+            End If
+            Dim resultNode = innerNode.With(leadingTrivia, trailingTrivia)
 
             resultNode = SimplificationHelpers.CopyAnnotations(castNode, resultNode)
 
             Return resultNode
 
         End Function
+
+        Private Function AddTriviaIfContainsAnyCommentOrLineContinuation(leadingTrivia As SyntaxTriviaList) As IEnumerable(Of SyntaxTrivia)
+            If leadingTrivia.ContainsAnyCommentOrLineContinuation Then
+                Return leadingTrivia
+            End If
+            Return New List(Of SyntaxTrivia)
```

Comment:
avoid these allocations.

Reply:
changed return to SyntaxTriviaList.Empty